### PR TITLE
Fix Qwen weight unrolling on MaxText

### PR DESCRIPTION
**Pull request by niting**
# Description

A recent change in MaxText caused the weight unrolling logic for Qwen models to be broken. This is needed if MaxText trainer runs against a VLLM sampler where scanned checkpoint weights are unscanned/unrolled before running on vllm. This adds back the calls to the unroll functions - which are still available.

We also fix some issues with dimension mismatches. We were using unpadded dimensions where padded dimensions should have been used and this causes responses from the model to be gibberish. 

# Tests

Ran Qwen3.5-35B on DeepSWE stack against TPU v5p-256. This seemed to output coherent responses.

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [X] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [X] I have necessary comments in my code, particularly in hard-to-understand areas.
- [X] I have run end-to-end tests tests and provided workload links above if applicable.
- [X] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

**Review by khatwanimohit — APPROVED** [review]

**Review by SurbhiJainUSC — APPROVED** [review]

File: src/maxtext/layers/moe.py [hunk 1]
```diff
@@ -689,9 +687,9 @@ def __init__(
       # During aqt convert state we delete kernel weight from params to save
       # memory. Instead they are retrieved from the tensors stored in the 'aqt'
       # collection.
-      self.wi_0 = jnp.zeros((num_experts, self.moe_expert_input_dim, intermediate_dim))
-      self.wi_1 = jnp.zeros((num_experts, self.moe_expert_input_dim, intermediate_dim))
-      self.wo = jnp.zeros((num_experts, intermediate_dim, self.moe_expert_input_dim))
+      self.wi_0 = jnp.zeros((num_experts, self.moe_expert_input_dim, moe_intermediate_dim))
+      self.wi_1 = jnp.zeros((num_experts, self.moe_expert_input_dim, moe_intermediate_dim))
+      self.wo = jnp.zeros((num_experts, self.intermediate_dim, self.moe_expert_input_dim))
```
Comment (YixuanWang-99) [inline src/maxtext/layers/moe.py:692]: Do you think wo should be `self.wo = jnp.zeros((num_experts, moe_intermediate_dim, self.moe_expert_input_dim))`? 

I thought the down-projection kernel wo maps from the intermediate activation space back to moe_expert_input_dim. Its contracting dimension must equal the output dimension of wi (i.e., moe_intermediate_dim).

File: src/maxtext/layers/moe.py [hunk 2]
```diff
@@ -708,7 +706,7 @@ def __init__(
               self.rngs.params(),
               (
                   self.num_experts,
-                  moe_intermediate_dim,
+                  self.intermediate_dim,
                   self.moe_expert_input_dim,
```
Comment (YixuanWang-99) [inline src/maxtext/layers/moe.py:710]: Similar to above, might use `moe_intermediate_dim` instead

File: src/maxtext/layers/moe.py
Comment (YixuanWang-99) [inline src/maxtext/layers/moe.py:684]: we defined the moe_intermediate_dim here

**Comment by YixuanWang-99** [comment]
I am bit worried about the scenario that `padded_base_moe_mlp_dim` is explicitly set (e.g., padding intermediate dimensions like 1408 to 1536 to satisfy GMM v2 TPU kernel alignment requirements). But we can deprioritize the concerns for now

**Review by YixuanWang-99 — APPROVED** [review]

File: src/maxtext/layers/moe.py (hunk 1 above)
Comment (khatwanimohit) [inline src/maxtext/layers/moe.py:692]: I think we can remove these set of changes cause they are under quantization block and that too for aqt quantization, we'll not fall under these conditions for RL workload
